### PR TITLE
[wordpress__data] Updated typings for version 6.0.0

### PR DESCRIPTION
**Pull request by marekdedic**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/WordPress/gutenberg/blob/trunk/packages/data/CHANGELOG.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by marekdedic** [comment]
Ok, I don'ŧ know why, but tests for multiple WP packages are broken. Should I try to fix them in this PR? They were broken before I started doing anything, however, I'm okay with fixing them - but that would make this PR touch several packages :/

**Comment by marekdedic** [comment]
This is probably related to #59550...

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## wordpress__data ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__data@latest/))
was missing the following properties:
1. AsyncModeProvider
1. controls
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f8172a7d05557014e1545f6feda08811d19604a9
</p>

**Comment by marekdedic** [comment]
Seems like #59550 fixed the CI issues :tada:

File: types/wordpress__data/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "private": true,
     "dependencies": {
-        "redux": "^4.0.1"
+        "redux": "^4.1.2"
```
Comment (sirreal) [inline types/wordpress__data/package.json:4]: [6.0 looks like it should be `^4.1.0`:](https://unpkg.com/browse/@wordpress/data@6.0.0/package.json)

```suggestion
        "redux": "^4.1.0"
```

**Review by sirreal — COMMENTED** [review]
I'm comfortable with redux upgrade and package bump to 6.0, I'd be happy to approve that.

Can you provide some details about the other changes? Did the API change? Were these missing? I haven't been following `@wordpress/data` closely so I'm not as comfortable reviewing those bits.

File: types/wordpress__data/package.json (hunk 1 above)
Comment (marekdedic) [reply types/wordpress__data/package.json]: Thanks, you are correct

**Comment by marekdedic** [comment]
Hi,
thanks for the correction in package version. As for the API change, see the [changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/CHANGELOG.md), the changes are done based on that and the package source.

**Review by rbuckton — APPROVED** [review]

**Comment by marekdedic** [comment]
Ready to merge

**Pull request merged**